Added external feed trigger docs (#2218)
OctopusDeploy/docs: src/pages/docs/projects/project-triggers/external-feed-triggers.md

Click the **ADD TRIGGER** on the right-hand side of the page, and select **External feed**

:::figure
![Project triggers list](/docs/projects/project-triggers/images/add-trigger-popup.png)
:::

Enter a name and description for your trigger.  The name should be short, memorable, and unique. Example: Create a release when nginx is updated in Docker.

## Channels and lifecycles

If your project contains multiple [channels](/docs/releases/channels), you have the option of selecting which channel this trigger will apply to.  Any pushed packages must satisfy the selected channel's [versioning rules](/docs/releases/channels#version-rules) to trigger a release creation. The releases created by the trigger will use this channel.

Unlike the existing [Automatic Release Creation](/docs/projects/project-triggers/automatic-release-creation) feature, you can create multiple external feed triggers per channel.  This can enable you to, for example, have your external container dependencies in a separate trigger to that used for your internal container dependencies.

:::figure
![Channel selection](/docs/projects/project-triggers/images/external-trigger-channel.png)
:::

> A preview of the [lifecycle](/docs/releases/lifecycles) is also shown, to show how releases created by the trigger will be deployed.
> 
> If you combine external feed triggers with the automatic deployment feature within [Lifecycles phases](/docs/releases/lifecycles/#Lifecycles-LifecyclePhases), whenever a package is pushed to a selected external feed, the trigger can create a release, and have it be automatically deployed.

Reviewer: This feels a bit stilted to me

```suggestion
A preview of the [lifecycle](/docs/releases/lifecycles) used by the selected channel is displayed. You can modify the [Lifecycles phases](/docs/releases/lifecycles/#Lifecycles-LifecyclePhases) to have a release created and deployed to selected environments whenever a new package is pushed.
```

Author: That's a good improvement.

## Trigger sources

> Any container images or helm charts referenced in your project's deployment process can be selected to trigger release creation.  Feeds or packages referenced using variable substitution are also options here, however they will only be used by the trigger if they are both evaluated as either container/helm chart repositories, and do not use variable unavailable at release creation time.  For example, using an environment name variable will not work, because that value is only available at deploy time.

Reviewer: nit about some of the text.

But also, the [writing docs](https://www.octopus.design/932c0f1a9/p/843fd5-documentation/t/43f893) seem to suggest that the 'edge case' of using variable feed types should go in an 'advance use case' section at the end.

```suggestion
Any container images or helm charts referenced in your project's deployment process can be selected to trigger release creation.  Feeds or packages referenced using variable substitution are also options here, however they will only be used by the trigger if they are both evaluated as either container or helm chart repositories, and do not use a variable unavailable at release creation time.  For example, using an environment name variable will not work, because that value is only available at deploy time.
```

:::figure
![Package selection](/docs/projects/project-triggers/images/external-feed-trigger-packages.png)
:::

:::div{.hint}
If you have a chain of dependencies with your external feed packages, make sure your trigger uses the package which will be pushed to its repository last.

> The release number that is created is guided by the Release Versioning settings under **Deployments Settings**. It will use the rules defined there.

Reviewer: This feels to me like it should be a separate hint and not in this section.

:::

## Troubleshooting {#ExternalFeedTriggers-Troubleshooting}

When you are using external feed triggers there are a few reasons why a release may not be created successfully. Take some time to consider the following troubleshooting steps:

1. **Inspect the task list** for errors in the **Task** menu - Octopus will log the reason why external feed triggers failed as errors or warnings.  Note that external feed triggers are system tasks, and do not display in the list by default.  Use the **SHOW ADVANCED FILTERS** option and select **Include system tasks** to show them.

> 2. Ensure you are pushing the package to a **supported external feed type**. While capability has been verified against most major docker providers, compatibility is not guaranteed - please contact octopus support if you encounter any problems. 

Author: ```suggestion
2. Ensure you are pushing the package to a **supported external feed type**. While capability has been verified against most major docker providers, compatibility is not guaranteed - please contact Octopus Deploy support if you encounter any problems. 
```

3. Ensure that packages in the external feed match the [channel rules](/docs/releases/channels#version-rules) if defined for the trigger's channel (or the default channel if your project doesn't have multiple channels).  **Triggers will only create a new release if the packages match channel rules.**

4. Ensure you are pushing a **new version** of the package - Octopus will not create a release where the package has already been used for creating a release.

> 5. Ensure you are pushing a package that Octopus will consider as the **latest available package**.  The trigger's version evaluator uses SemVer, and will not triggers off image tags such as 'latest'.

Author: ```suggestion
5. Ensure you are pushing a package that Octopus will consider as the **latest available package**.  The trigger's version evaluator uses SemVer, and will not trigger off image tags such as 'latest'.
```

> 6. Make sure that the feed and package references **only use variables which are able to be evaluated** at release creation time.

Author: ```suggestion
6. Make sure that the feed and package references only use variables which are **able to be evaluated at release creation time.**
```

Reviewer: We should specify what this means, eg. "doesn't evaluate environment variables"

Your existing [lifecycles](/docs/releases/lifecycles/) will then promote that release through your [environments](/docs/infrastructure/environments) or [tenants](/docs/tenants), just like it does currently. If your lifecycle uses automatic release progression, then you've just set up a Continuous Delivery pipeline without explicitly letting Octopus know about your application changes! 

The details of these container images and Helm Charts are already known in Octopus. This means we can use the registry locations, image names, chart names, and credentials to do this monitoring, without adding or maintaining this information anywhere else.

## Getting started {#ExternalFeedTriggers-GettingStarted}

Navigate to your project’s triggers page by selecting Projects and clicking on the project you are working with. Click **Triggers** option on the left, under **Deployments**. You are now presented with an empty triggers list. 

> Click the **ADD TRIGGER** on the right-hand side of the page, and select **External feed**

Reviewer 2: Missing a . at the end of the sentence 🙂

:::figure
![Project triggers list](/docs/projects/project-triggers/images/add-trigger-popup.png)
:::

> Enter a name and description for your trigger. The name should be short, memorable, and unique. Example: Create a release when nginx is updated in Docker.

Reviewer 2: Could we make the example name a bit shorter? Something like _Nginx Docker Update_.
